fabric-recipe-api-v1/src/main/java/net/fabricmc/fabric/impl/recipe/ingredient/CustomIngredientImpl.java

```diff
@@ -114,11 +133,10 @@ public void write(PacketByteBuf buf) {
 	}
 
 	@Override
-	public JsonElement toJson() {
-		JsonObject json = new JsonObject();
-		json.addProperty(TYPE_KEY, customIngredient.getSerializer().getIdentifier().toString());
-		customIngredient.getSerializer().write(json, coerceIngredient());
-		return json;
+	public JsonElement toJson(boolean bl) {
+		// TODO do we need to care about bl? I think we do.
```

Comment:
Note: I think this boolean has something to do with allowing empty stacks. Ingridents have 2 codecs, I think we need to also have 2?

Reply:
Correct, the boolean decides whether `minecraft:air` is allowed

Reply:
They changed it to allow an empty array instead, rather than `minecraft:air` in [1.20 pre-release 6](https://www.minecraft.net/en-us/article/minecraft-1-20-pre-release-5). `minecraft:air` is no longer allowed.